### PR TITLE
[stable/velero] documentation and cleanup

### PR DESCRIPTION
**Pull request by skriss**
#### What this PR does / why we need it:
- Adds/updates documentation
- Fixes a handful of minor issues

#### Which issue this PR fixes
N/A

#### Special notes for your reviewer:
cc @unguiculus @domcar @nrb

Thanks all for all the work on the Velero chart so far! I've taken a pass through it and added/updated a bunch of documentation and fixed a couple small issues that I saw. There shouldn't be any breaking changes in this PR.

Heads-up that we'll probably be submitting another PR in the next couple weeks to get ready for the Velero v1.0 release -- and I expect that PR will contain some breaking chart changes. Some of the things I've seen so far that I'd like to update:
- as of v1.0 we will support sourcing Azure credentials from a file, so we can remove the special-casing for Azure/secrets
- remove the `config` section from `values.yaml` since that's a vestige of an old CRD that no longer exists; move the contents out into appropriate places
- remove `config.provider` field since it's redundant with `config.backupStorageLocation.name` and `config.volumeSnapshotLocation.name`
- update the `.name` fields mentioned in the previous bullet to be `.provider`
- I'd love to add support for creating the `cloud-credentials` secret from a file, a la https://helm.sh/docs/chart_template_guide/#configmap-and-secrets-utility-functions

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Review by jlegrone — COMMENTED** [review]
/ok-to-test

File: stable/velero/templates/volumesnapshotlocation.yaml
Comment (jlegrone) [inline stable/velero/templates/volumesnapshotlocation.yaml]: This looks like a breaking change. Perhaps we could hold off on it until the next major release, or add some logic to the template to handle either value.

Also if this is only required for Azure, then it would be possible to make missing values a templating error rather than a potential runtime error (and give chart users better feedback) by doing something like:

```yaml
{{- if eq .name "azure" }}
  apiTimeout: {{ required "configuration.volumeSnapshotLocation.config.apiTimeout is required for azure volumeSnapshotLocation" .apiTimeout }}
{{- end }}
```

File: stable/velero/templates/volumesnapshotlocation.yaml
Comment (jlegrone) [inline stable/velero/templates/volumesnapshotlocation.yaml]:
```suggestion
   provider: {{ .name }}
```

File: stable/velero/templates/volumesnapshotlocation.yaml
Comment (skriss) [reply stable/velero/templates/volumesnapshotlocation.yaml]: fair point - the `values.yaml` and documentation use `apiTimeout`, so I was updating it to match, but I suppose it should go the other way around for this PR. 

I'm not familiar with the `required` syntax, but I can look into it.

File: stable/velero/templates/volumesnapshotlocation.yaml
Comment (skriss) [reply stable/velero/templates/volumesnapshotlocation.yaml]: fixed in latest commit

File: stable/velero/templates/volumesnapshotlocation.yaml
Comment (skriss) [reply stable/velero/templates/volumesnapshotlocation.yaml]: reverted breaking change in latest commit, and updated docs & values.yaml to reflect lowercased field name

**Comment by skriss** [comment]
I believe in https://github.com/helm/charts/pull/12031 some manual cleanup of the test env was required, related to CRDs, but I don't know the details

**Comment by jlegrone** [comment]
@skriss it looks like other charts have fixed this issue by executing a CRD cleanup job in CI (see [discussion](https://kubernetes.slack.com/archives/C6E3XH1ED/p1556041680040900)). I'm not sure who would have access to manually delete CRDs in the test cluster.

The idea is to create a post-delete hook which runs a [job that deletes each CRD managed by the chart](https://github.com/helm/charts/blob/3caf789dfe65a8416be6c82b93c4bf1df56fedc1/stable/prometheus-operator/templates/prometheus-operator/cleanup-crds.yaml). We won't want that to be the default behavior, however, so a new values file would be needed in the `ci` directory (something like `stable/velero/ci/test-values.yaml`) to enable cleanup in CI only.

**Comment by skriss** [comment]
Well, I certainly wasn't planning on signing up for fixing CI as part of adding comments to the chart :) I literally just started working with Helm yesterday, so I'm gonna have to ramp up on all this some more to possibly be able to help. Is it a blocker for this PR, given that we've had previous merges to this chart in the current state?

**Comment by skriss** [comment]
@unguiculus I know you fixed this CI issue for the previous PR to the chart - can you help here?

**Comment by skriss** [comment]
Great, looks like CI is passing now

**Comment by nrb** [comment]
@jlegrone Can you expand on this?

>The idea is to create a post-delete hook which runs a job that deletes each CRD managed by the chart. We won't want that to be the default behavior however

Why is cleaning up the CRDs managed by the chart not desirable as the default behavior? Is there some documentation I can read up on about this?

File: stable/velero/Chart.yaml
Comment (jlegrone) [inline stable/velero/Chart.yaml]: Sorry I didn't catch this before. Documentation-only changes should just be a patch version bump. Due to the issues with CI I would be okay merging as is for now but I would like to get a second opinion first.

```suggestion
version: 1.0.1
```

Comment (unguiculus) [reply stable/velero/Chart.yaml]: Agreed. Let's bump the patch version.

**Review by jlegrone — APPROVED** [review]
/lgtm

**Comment by skriss** [comment]
added signoff

**Comment by jlegrone** [comment]
> Why is cleaning up the CRDs managed by the chart not desirable as the default behavior? Is there some documentation I can read up on about this?

@nrb deleting CRDs is potentially dangerous, since all custom resources tied to a CRD are garbage collected by kubernetes after deletion. This was part of the reason for managing CRDs with a custom helm hook: https://github.com/helm/helm/issues/4489#issuecomment-414135147.

**Comment by skriss** [comment]
🎉

**Pull request merged**
